src/modules/drive/Toolbar/components/LeaveSharedDriveButtonItem.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { useNavigate } from 'react-router-dom'
+
+import { useClient } from 'cozy-client'
+import ActionsMenuItem from 'cozy-ui/transpiled/react/ActionsMenu/ActionsMenuItem'
+import Icon from 'cozy-ui/transpiled/react/Icon'
+import TrashIcon from 'cozy-ui/transpiled/react/Icons/Trash'
+import ListItemIcon from 'cozy-ui/transpiled/react/ListItemIcon'
+import ListItemText from 'cozy-ui/transpiled/react/ListItemText'
+import { useAlert } from 'cozy-ui/transpiled/react/providers/Alert'
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+
+const LeaveSharedDriveButtonItem = ({ files }) => {
+  const { t } = useI18n()
+  const client = useClient()
+  const navigate = useNavigate()
+  const { showAlert } = useAlert()
+  const handleClick = async () => {
+    const file = files[0]
+    const sharingId = findSharingId(file)
+    if (sharingId) {
+      await client.collection('io.cozy.sharings').revokeSelf({ _id: sharingId })
+      showAlert({
+        message: t('Files.share.revokeSelf.success'),
+        severity: 'success',
+      })
+      navigate('/sharings')
+    }
+  }
+
+  return (
+    <ActionsMenuItem onClick={handleClick}>
+      <ListItemIcon>
+        <Icon icon={TrashIcon} className="u-error" />
+      </ListItemIcon>
+      <ListItemText primary={t('toolbar.menu_leave_shared_drive')} />
+    </ActionsMenuItem>
+  )
+}
+
+export default LeaveSharedDriveButtonItem
+
+function findSharingId(folder) {
+  const references = folder.relationships.referenced_by.data
+  const sharingId = references.find(
+    (ref) => ref.type === 'io.cozy.sharings'
+  )?.id
+  return sharingId
+}
```

Comment:
nit: Maybe this method should be somewhere else? Or maybe we already have something like this?